src/internal_events/remap.rs

```diff
@@ -0,0 +1,37 @@
+use super::InternalEvent;
+use metrics::counter;
+
+#[derive(Debug)]
+pub struct RemapEventProcessed;
+
+impl InternalEvent for RemapEventProcessed {
+    fn emit_metrics(&self) {
+        counter!("events_processed", 1,
+            "component_kind" => "transform",
+            "component_type" => "remap",
+        );
+    }
+}
+
+#[derive(Debug)]
+pub struct RemapFailedMapping {
+    pub error: String,
+}
+
+impl InternalEvent for RemapFailedMapping {
+    fn emit_logs(&self) {
+        warn!(
+            message = "Mapping failed with event",
```

Comment:
This doesn't confirm with https://github.com/timberio/vector/blob/master/rfcs/2020-03-17-2064-event-driven-observability.md, please use a `; discarding event.` at the end.

Reply:
Resolved in 66ea912.

Although thinking about this, I wonder if we instead want to use a fixed `Mapping failed with event.` message, and then have a `dropped` boolean field attached to the log?